File: TestAndroidV2/app/src/main/java/br/com/altran/santander/davidmelo/ui/adapter/StatementsAdapter.java
package br.com.altran.santander.davidmelo.ui.adapter;

import android.annotation.SuppressLint;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import br.com.altran.santander.davidmelo.R;
import br.com.altran.santander.davidmelo.model.StatementList;

import java.text.NumberFormat;
import java.util.ArrayList;

public class StatementsAdapter extends RecyclerView.Adapter<StatementsAdapter.ViewHolder> {
    private ArrayList<StatementList> list;

    public StatementsAdapter(ArrayList<StatementList> list) {
        this.list = list;
    }

    @NonNull
    @Override
    public StatementsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new ViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_bill, viewGroup, false));
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull StatementsAdapter.ViewHolder holder, int position) {
        holder.title.setText(list.get(position).getTitle());
        holder.type.setText(list.get(position).getDesc());
        holder.date.setText(list.get(position).getDate());
        holder.value.setText(NumberFormat.getCurrencyInstance().format(list.get(position).getValue()) + "");
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView title;
        TextView type;
        TextView date;
        TextView value;

        ViewHolder(View v) {
            super(v);
                value = v.findViewById(R.id.value_price);
                title = v.findViewById(R.id.payment_text);
                type = v.findViewById(R.id.type_text);
                date = v.findViewById(R.id.payment_date);


        }
    }
}

File: app/src/main/java/com/example/otavioaugusto/testesantander/adapters/StatementsAdapter.kt
package com.example.otavioaugusto.testesantander.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.otavioaugusto.testesantander.R
import com.example.otavioaugusto.testesantander.model.StatementListItem
import kotlinx.android.synthetic.main.item_statements.view.*
import java.text.FieldPosition

class StatementsAdapter(val lista:List<StatementListItem>, val contexto:Context) : RecyclerView.Adapter<StatementsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_statements, parent, false)
        val view = ViewHolder(v)
        return  view

    }

    override fun getItemCount(): Int {
       return lista.size

    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        holder.title.text = lista[position].title
        holder.desc.text = lista[position].desc
        holder.date.text = lista[position].date
        holder.value.text = lista[position].value.toString()

    }


    class ViewHolder(itemView:View) : RecyclerView.ViewHolder(itemView){

        val title = itemView.txtItemTitle
        val desc = itemView.txtItemDescricao
        val date = itemView.txtItemDate
        val value = itemView.txtItemValue

    }
}